enclave-tls/src/verifiers/internal/etls_enclave_verifier_load_single.c

```diff
@@ -0,0 +1,82 @@
+/* Copyright (c) 2020-2021 Alibaba Cloud and Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <string.h>
+#include <dlfcn.h>
+#include <enclave-tls/err.h>
+#include <enclave-tls/log.h>
+#include "internal/core.h"
+#include "internal/verifier.h"
+
+#define PATTERN_PREFIX "libverifier_"
+#define PATTERN_SUFFIX ".so"
+
+enclave_tls_err_t etls_enclave_verifier_load_single(const char *fname)
+{
+	ETLS_DEBUG("loading the enclave verifier instance '%s' ...\n", fname);
+
+	/* Check whether the filename pattern matches up libenclave_verifier_<name>.so */
+	if (strlen(fname) <= strlen(PATTERN_PREFIX) + strlen(PATTERN_SUFFIX) ||
+	    strncmp(fname, PATTERN_PREFIX, strlen(PATTERN_PREFIX)) ||
+	    strncmp(fname + strlen(fname) - strlen(PATTERN_SUFFIX), PATTERN_SUFFIX,
+		    strlen(PATTERN_SUFFIX))) {
+		ETLS_ERR("The filename pattern of '%s' NOT match " PATTERN_PREFIX
+			 "<name>" PATTERN_SUFFIX "\n",
+			 fname);
+		return -ENCLAVE_TLS_ERR_INVALID;
+	}
+
+	char realpath[strlen(ENCLAVE_VERIFIERS_DIR) + strlen(fname) + 1];
+	sprintf(realpath, "%s%s", ENCLAVE_VERIFIERS_DIR, fname);
+
+	void *handle = dlopen(realpath, RTLD_LAZY);
+	if (!handle) {
+		ETLS_ERR("failed on dlopen(): %s\n", dlerror());
+		return -ENCLAVE_TLS_ERR_DLOPEN;
+	}
+
+	size_t name_len = strlen(fname) - strlen(PATTERN_PREFIX) - strlen(PATTERN_SUFFIX);
+	char name[name_len + 1];
+	strncpy(name, fname + strlen(PATTERN_PREFIX), name_len);
+	name[name_len] = '\0';
+
+	unsigned int i = 0;
+	enclave_verifier_opts_t *opts = NULL;
+	for (i = 0; i < registerd_enclave_verifier_nums; ++i) {
+		opts = enclave_verifiers_opts[i];
+
+		if (!strcmp(name, opts->name))
+			break;
+	}
+
+	if (i == registerd_enclave_verifier_nums) {
+		ETLS_ERR("the enclave verifier '%s' is not registered yet\n", name);
```

Comment:
Change to `the enclave verifier '%s' failed to register`.

Reply:
done